haystack/document_stores/elasticsearch/es7.py

```diff
@@ -178,6 +178,13 @@ def __init__(
             batch_size=batch_size,
         )
 
+        if self.server_version[0] != 7:
```

Comment:
Can you isolate this with expected version being a parameter and then reuse it for 8?